lib/cretonne/src/dominator_tree.rs

```diff
@@ -37,6 +37,8 @@ pub struct DominatorTree {
 
     // Scratch memory used by `compute_postorder()`.
     stack: Vec<Ebb>,
+
+    valid: bool,
```

Comment:
What's the motivation for having a separate valid flag instead of just clearing the `nodes` when invalidating the tree?

This variable doesn't seem to be reset anywhere.

Reply:
What prompted me was the `empty` tests in dominator_tree.rs, topo_order.rs, and others, which construct empty functions and call various dominator/etc. functions on them. They do call `compute`, but there are no basic blocks, so `nodes` is empty. Would you prefer to change those tests to add an entry block?

The lack of a reset is a bug in my patch. It should be reset in `clear`.

Reply:
OK, that makes sense. Empty functions are a silly corner case, but let's just leave the `valid` flag in.